test/constructs/if_test.exs

```diff
@@ -66,4 +66,321 @@ defmodule Surface.Constructs.IfTest do
       compile_surface(code)
     end)
   end
+
+  describe "#if language structure" do
+    test "renders inner if condition is truthy" do
+      html =
+        render_surface do
+          ~H"""
+          <#if condition={true}>
+          <span>The inner content</span>
+          <span>with multiple tags</span>
+          </#if>
+          """
+        end
+
+      assert html =~ """
+             <span>The inner content</span>
+             <span>with multiple tags</span>
+             """
+    end
+
+    test "parser error message contains the correct line" do
+      code =
+        quote do
+          ~H"""
+          <#if condition={true}>
+            <span>The inner content
+          </#if>
+          """
+        end
+
+      message = ~S(code:2: expected closing tag for <span>)
+
+      assert_raise(Surface.Compiler.ParseError, message, fn ->
+        compile_surface(code)
+      end)
+    end
+
+    test "compile error message contains the correct line" do
+      code =
+        quote do
+          ~H"""
+          <#if condition={false}>
+            <ListProp prop="some string" />
+          </#if>
+          """
+        end
+
+      message =
+        ~S(code:2: invalid value for property "prop". Expected a :list, got: "some string".)
+
+      assert_raise(CompileError, message, fn ->
+        compile_surface(code)
+      end)
+    end
+  end
+
+  describe "#else language structure" do
+    test "renders inner if condition if condition is truthy" do
+      html =
+        render_surface do
+          ~H"""
+          <#if condition={true}>
+          <span>The inner content</span>
+          <span>with multiple tags</span>
+          <#else>
+          <span>The else content</span>
+          <span>with multiple tags</span>
+          </#if>
+          """
+        end
+
+      assert html =~ """
+             <span>The inner content</span>
+             <span>with multiple tags</span>
+             """
+    end
+
+    test "renders inner else condition is falsly" do
```

Comment:
small typo, should be `falsey` i think, maybe `falsy` - i'm honestly not sure which one 😂